SimCalorimetry/PhaseIAnalyzer/plugins/PhaseIAnalyzer.cc

```diff
@@ -0,0 +1,254 @@
+// -*- C++ -*-
+//
+// Package:    PhaseI/PhaseIAnalyzer
+// Class:      PhaseIAnalyzer
+//
+/**\class PhaseIAnalyzer PhaseIAnalyzer.cc PhaseI/PhaseIAnalyzer/plugins/PhaseIAnalyzer.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Dario Soldi
+//         Created:  Tue, 16 Jan 2018 11:56:05 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+//My includes
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CondFormats/EcalObjects/interface/EcalPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalPedestalsRcd.h"
+#include "CondFormats/DataRecord/interface/EcalIntercalibConstantsMCRcd.h"
+#include "CondFormats/EcalObjects/interface/EcalIntercalibConstantsMC.h"
+
+#include "PhaseIAnalyzer.h"
+
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "SimDataFormats/CaloHit/interface/PCaloHit.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "SimDataFormats/Track/interface/SimTrack.h"
+#include "CondFormats/EcalObjects/interface/EcalConstants.h"
+
+using namespace std;
+using namespace edm;
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+
+PhaseIAnalyzer::PhaseIAnalyzer(const edm::ParameterSet& iConfig) {
```

Comment:
if this class doesn't need to run in production jobs, I would move it to the test directory.
also, for plugin classes, it's recommended not to have a separate .h file (define the class in the .cc file). This is because plugins shouldn't be included by other code.